packages/swarm-components/src/NumericalInput.jsx

```diff
@@ -2,47 +2,185 @@
 import * as React from 'react';
 import Icon from './Icon';
 
+type Value = number | null;
 type Props = {
     /**
-     * Classname of wrapper, specified because it is not applied with other
+     * indicates whether the input is disabled
      */
-    className?: string,
+    disabled?: boolean,
     /**
-     * Whether the numerical input should be interactive.
+     * indicates whether there is an error on the input
      */
-    disabled?: boolean,
+    error?: React$Node,
     /**
-     * Whether the textarea renders an error state.
+     * id of the input
      */
-    error?: boolean,
+    id: string,
     /**
-     * Value for textarea.
+     * the maximum integer allowed
      */
-    value?: string
-}
+	max?: number,
+    /**
+     * the minimum integer allowed
+     */
+    min?: number,
+    /**
+     * name of the input, used with labels
+     */
+    name: string,
+    /**
+     * Required change handler with Value, not event
+     */
+    onChange: Value => void,
+    /**
+     * funcitonality invoked when the form field is blurred
+     */
+    onBlur: (SyntheticInputEvent<HTMLInputElement>) => void,
+    /**
+     * The amount the input will increment or decrement when using keyboard interactions
+     */
+    step: number,
+    /**
+     * The value of the field
+     */
+	value: Value,
+};
+
+type State = {
+	isFieldFocused: boolean,
+	value: Value,
+};
 
 const getStatus = (props: Props): string => {
-    return props.disabled ? 'disabled' : (props.error ? 'error' : 'default');
+	return props.disabled ? 'disabled' : props.error ? 'error' : 'default';
 };
 
-const getCharacterCount = (value: string = '') => value.length;
+export class NumericalInput extends React.Component<Props> {
+	static defaultProps = {
+		step: 1,
+		min: 0,
+	};
 
-const NumericalInput = (props: Props): React.Element<'textarea'> => {
+	state = {
+		value: this.props.value || this.props.value === 0 ? this.props.value : null,
+		isFieldFocused: false,
+	};
 
-    const { className = '', ...other } = props;
+    // internal value updater when passed a value via props
+	static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+		const isNewValue = nextProps.onChange && nextProps.value !== prevState.value;
```

Comment:
@hsbacot and I talked about this, and the feeling was that, since the exported library doesn't export its types, there's no way to enforce that consumers supply an `onChange` at all, so this is defensive programming to allow a consumer to just supply the 'initial value' - an unlikely use case, but one supported by the previous input. There might be a better way of enforcing it, but ultimately we should export our flow types (or convert to typescript)